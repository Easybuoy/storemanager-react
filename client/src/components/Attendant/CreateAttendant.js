import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';

import { createAttendant } from '../../actions/attendantActions';
import Loading from '../Common/Loading';

class CreateAttendant extends Component {
constructor(){
  super();
  this.state = {
    name: '',
    email: '',
    password: '',
    userimage: '',
    type: ''
  };
}

onChange = (event) => {
  this.setState({ [event.target.name]: event.target.value });
}

onImageChange = (event) => {
  this.setState({ userimage: event.target.files[0] });
}

onSubmit = (e) => {
  e.preventDefault();
  const { name, email, password, userimage, type } = this.state;

  const userData = {
    name,
    email,
    password,
    userimage,
    type
  };

  this.props.createAttendant(userData);
}


  render() {
    const { errors} = this.props;
    const { isAttendantCreated, loading } = this.props.attendants;

    if (loading) {
      return (
          <div>
          <Loading />
          </div>
      )
  }
    console.log(errors)
    if (Object.keys(errors).length > 0) {
      if (errors.email) {
        toast.error(errors.email);
      }

      if (errors.password) {
        toast.error(errors.password);
      }
    }

    if (isAttendantCreated) {
      toast.success(isAttendantCreated);
    }

    return (
      <div id="main">
        <div className="container">
            <section className="topmargin">
                <div className="container">
                        <div id="create_sales_assistant">
                                <h1 className="text-center">Create Sales Attendant</h1>
                        <form onSubmit={this.onSubmit}>
                                <h2>Name</h2>
                                <input type="text" name="name" id="name" placeholder="Enter Name" onChange={this.onChange} required/>
                                <h2>Email</h2>
                                <input type="email" name="email" id="email" placeholder="Enter Email" onChange={this.onChange} required/>
                                <h2>Password</h2>
                                <input type="password" name="password" id="password" placeholder="Enter Password" onChange={this.onChange} required/> <br />
                                <h2>User Image</h2>
                                <input type="file" id="userimage" name="upload" onChange={this.onImageChange}/>
                                <h2>Type</h2>
                                <select id="type" name="type" onChange={this.onChange} required>
                                    <option value="">Select Role</option>
                                    <option value="1">Admin</option>
                                    <option value="2">Attendant</option>
                                </select> <br /> <br />
                                <input type="submit" name="" id="createattendant" className="button_1" value="Create Sales Assistant" />
                                
                            </form>   
                    </div>
                </div>
            </section>
      </div>
    </div>
    )
  }
}

const mapStateToProps = state => ({
  attendants: state.attendants,
  errors: state.errors
});

export default connect(mapStateToProps, { createAttendant })(CreateAttendant);