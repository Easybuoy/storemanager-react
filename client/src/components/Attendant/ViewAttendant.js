import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { viewAttendants } from '../../actions/attendantActions';
import Loading from '../Common/Loading';

class ViewAttendant extends Component {
    componentDidMount() {
        this.props.viewAttendants();
    }

    render() { console.log(this.props)
        const { loading, attendants } = this.props.attendants;

        if (loading) {
            return (
                <div>
                    <Loading />
                </div>
            )
        }

        if (attendants) {
            return (
            <div id="main" className="topmargin">
            <h1 className="text-center">Sale Attendants</h1>
                <div className="container topmargin">
                <div className="cardgroup">
                {
                    attendants.map((attendant, key) => {
                        const { name, email, userimage } = attendant; 
                        const attendantimage = `https://store--manager.herokuapp.com/${userimage}`;
                        return (
                            <div key={key} className="card"> 
                                <a><img src={attendantimage} className="cardimg" /></a>
                            
                            <div className="text-center cardbody" >
                              <h3 id="productname">Name: {name}</h3>
                              <p>Email: {email} </p>
                              <p>Attendance: -</p>
                              <p>Product Sold: -</p>
                              <button className="button_3" >Promote</button>
                              <button className="button_2" >Delete</button>
                            </div>
                            </div>
                        )
                    })
                }
                    </div>
                </div>
            </div>
            )
        }
        return (
            <div id="main" className="topmargin">
            <h1 className="text-center">Sale Attendants</h1>
              <div className="card">
                <a><img src="${attendant_image}" className="cardimg" /></a>
                            
              <div className="text-center cardbody" >
                <h3 id="productname">Name: }</h3>
                <p>Email: </p>
                <p>Attendance: -</p>
                <p>Product Sold: -</p>
                <button className="button_3" >Promote</button>
                <button className="button_2" >Delete</button>
              </div>
                            
              </div>
            </div>
          )
    }
}

ViewAttendant.propTypes = {
    viewAttendants: PropTypes.func.isRequired,
    attendants: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
    attendants: state.attendants
});

export default  connect(mapStateToProps, { viewAttendants })(ViewAttendant);