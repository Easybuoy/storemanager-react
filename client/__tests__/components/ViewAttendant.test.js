import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { ViewAttendant } from '../../src/components/Attendant/ViewAttendant';

describe('<ViewAttendant />', () => {
    const props = {
        viewAttendants: jest.fn(),
        deleteAttendant: jest.fn(),
        attendants: jest.mock(),
      };

    it('renders the ViewAttendant component correctly', () => {
        const wrapper = shallow(<ViewAttendant {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should call the mock viewAttendant function', () => {
        const wrapper = shallow(<ViewAttendant {...props} />)
 
         expect(props.viewAttendants).toBeCalled();
         expect(props.viewAttendants).toHaveReturned();
        });

        it('should render salesRecord if sales array is not empty', () => {
            props.attendants.attendants = [{id: 1, name: 'iphone'}];
            const wrapper = shallow(<ViewAttendant {...props} />)

            expect(props.viewAttendants).toBeCalled();
             expect(props.viewAttendants).toHaveReturned();
            });

            it('should render Loading component if loading props is true', () => {
                props.attendants.loading = true;
                const wrapper = shallow(<ViewAttendant {...props} />)
    
                expect(props.viewAttendants).toBeCalled();
                 expect(props.viewAttendants).toHaveReturned();
                });
                
            it('should render toast if attendantDeleted props is true', () => {
                    props.attendants.attendantDeleted = true;
                    const wrapper = shallow(<ViewAttendant {...props} />)
        
                    expect(props.viewAttendants).toBeCalled();
                     expect(props.viewAttendants).toHaveReturned();
                    });
            
                    it('should call the mock deleteAttendant function', () => {
                        const wrapper = shallow(<ViewAttendant {...props} />)
                 
                        Object.defineProperty(window.confirm, 'confirm', {
                            configurable: true,
                          });
                          window.confirm = jest.fn();
                
                          
                         const id = 1;
                         sinon.spy(wrapper.instance(), 'deleteAttendant');
                         wrapper.instance().deleteAttendant(id);
                         expect(wrapper.instance().deleteAttendant.calledOnce)
                           .toEqual(true);
                        });
});