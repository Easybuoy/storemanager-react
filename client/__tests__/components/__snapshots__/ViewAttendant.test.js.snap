// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewAttendant /> renders the ViewAttendant component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewAttendant
    attendants={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "isMockFunction": [Function],
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
    deleteAttendant={[MockFunction]}
    viewAttendants={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loading />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loading />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
